<?xml version="1.0" ?>

<launch>
    <arg name="robot_name"              default="Gripper_1"/>
    <param name="offset_file_path"      value="$(find test_gripper_manager)/config/offset.yaml"/>
    <param name="robot_file_path"       value="$(find test_gripper_manager)/config/$(arg robot_name).robot"/>
    <param name="init_file_path"        value="$(find test_gripper_manager)/config/dxl_init.yaml"/>

    <!-- TEST-Gripper Manager -->
    <node name="$(arg robot_name)_manager" pkg="test_gripper_manager" type="test_gripper_manager" output="screen">
      <param name="robot_name" value="$(arg robot_name)"/>
      <remap from="/robotis/enable_ctrl_module" to="/$(arg robot_name)/enable_ctrl_module"/>
      <remap from="/robotis/sync_write_item" to="/$(arg robot_name)/sync_write_item"/>
      <remap from="/robotis/test_gripper/movement_done" to="/$(arg robot_name)/test_gripper/movement_done"/>
      <remap from="/robotis/status" to="/$(arg robot_name)/status"/>
    </node>
    
    <!-- TEST-Gripper GUI -->
    <node name="test_gripper_gui" pkg="test_gripper_gui" type="test_gripper_gui" output="screen">
      <param name="robot_name" value="$(arg robot_name)"/>
      <remap from="/robotis/status" to="/$(arg robot_name)/status"/>
    </node>

</launch>
